SYCL/Plugin/level_zero_sub_sub_device.cpp

```diff
@@ -25,89 +25,65 @@ using namespace std::chrono;
-    buffer_host2[i] = static_cast<float>(random_float());
-    buffer_host3[i] = static_cast<float>(random_float());
+    buffer_hosts.push_back(buffer_host);
+    buffer_devices.push_back(buffer_device);
+    queues.push_back(temp);
   }
 
-  queues[0].memcpy(buffer_device0, buffer_host0, N * sizeof(float)).wait();
-  queues[1].memcpy(buffer_device1, buffer_host1, N * sizeof(float)).wait();
-  queues[2].memcpy(buffer_device2, buffer_host2, N * sizeof(float)).wait();
-  queues[3].memcpy(buffer_device3, buffer_host3, N * sizeof(float)).wait();
-
+  // Run workload
   for (auto m = 0; m < INTER_NUM; ++m) {
     for (int k = 0; k < KERNEL_NUM; ++k) {
-      auto event0 = queues[0].submit([&](handler &h) {
-        h.parallel_for<class kernel0>(
-            nd_range<1>(range<1>{global_range}, range<1>{local_range}),
-            [=](nd_item<1> item) {
-              int i = item.get_global_linear_id();
-              buffer_device0[i] = buffer_device0[i] + float(2.0);
-            });
-      });
-      auto event1 = queues[1].submit([&](handler &h) {
-        h.parallel_for<class kernel1>(
-            nd_range<1>(range<1>{global_range}, range<1>{local_range}),
-            [=](nd_item<1> item) {
-              int i = item.get_global_linear_id();
-              buffer_device1[i] = buffer_device1[i] + float(2.0);
-            });
-      });
-      auto event2 = queues[2].submit([&](handler &h) {
-        h.parallel_for<class kernel2>(
-            nd_range<1>(range<1>{global_range}, range<1>{local_range}),
-            [=](nd_item<1> item) {
-              int i = item.get_global_linear_id();
-              buffer_device2[i] = buffer_device2[i] + float(2.0);
-            });
-      });
-      auto event3 = queues[3].submit([&](handler &h) {
-        h.parallel_for<class kernel3>(
+      for (int j = 0; j < queues.size(); j++) {
+
+        queue current_queue = queues[j];
+        float *buffer_device = buffer_devices[j];
+
+        auto event0 = current_queue.parallel_for<>(
             nd_range<1>(range<1>{global_range}, range<1>{local_range}),
             [=](nd_item<1> item) {
               int i = item.get_global_linear_id();
-              buffer_device3[i] = buffer_device3[i] + float(2.0);
+              buffer_device[i] = buffer_device[i] + float(2.0);
             });
-      });
+      }
     }
-    queues[0].wait();
-    queues[1].wait();
-    queues[2].wait();
-    queues[3].wait();
-  }
-
-  free(buffer_host0, queues[0]);
-  free(buffer_device0, queues[0]);
-
-  free(buffer_host1, queues[1]);
-  free(buffer_device1, queues[1]);
 
-  free(buffer_host2, queues[2]);
-  free(buffer_device2, queues[2]);
+    for (auto q : queues)
+      q.wait();
+  }
 
-  free(buffer_host3, queues[3]);
-  free(buffer_device3, queues[3]);
+  // De-allocate buffer hosts and devices
+  for (int j = 0; j < queues.size(); j++) {
+    sycl::free(buffer_devices[j], queues[j]);
+    sycl::free(buffer_hosts[j], queues[j]);
+  }
```

Comment:
The code is trivial, no need to state the same in a comment.
```suggestion
  for (int j = 0; j < queues.size(); j++) {
    sycl::free(buffer_devices[j], queues[j]);
    sycl::free(buffer_hosts[j], queues[j]);
  }
```